spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/data/redis/RedisAutoConfigurationJedisTests.java

```diff
@@ -135,13 +144,32 @@ void testRedisConfigurationWithPool() {
 				});
 	}
 
-	@Test
-	void testRedisConfigurationWithTimeout() {
-		this.contextRunner.withPropertyValues("spring.redis.host:foo", "spring.redis.timeout:100").run((context) -> {
-			JedisConnectionFactory cf = context.getBean(JedisConnectionFactory.class);
-			assertThat(cf.getHostName()).isEqualTo("foo");
-			assertThat(cf.getTimeout()).isEqualTo(100);
-		});
+	@ParameterizedTest(name = "{0}")
+	@MethodSource("timeoutArguments")
+	void testRedisConfigurationWithTimeout(List<String> properties, long readTimeout, long connectionTimeout) {
```

Comment:
Rather than a parametrised test, I think we can go with two tests. One that doesn't set anything and make sure defaults are properly honoured and one that sets both the timeout and the connect timeout and check the override has been customized.

Reply:
done